toolkit/tools/imagegen/configuration/veritydisk.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parser for the image builder's configuration schemas.
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+)
+
+// ReadOnlyVerityRoot controls DM-Verity read-only filesystems which will be mounted at startup
+// It will create a verity disk from the partition mounted at "/". The verity data is stored as
+// part of the image's initramfs.
+// - Enable: Enable dm-verity on the root filesystem and add the root hash to the
+//     initramfs
+// - Name: Custom name for the mounted root (default is "verity_root_fs")
+// - ErrorCorrectionEnable: Enable Reed-Solomon forward error correction of read-only data and
+//     add the FEC data to the initramfs
+// - ErrorCorrectionEncodingRoots: Increase overhead to increase resiliency, default is 2
+//     encoding bytes per 255 bytes of real data) giving 0.8% overhead ( RS(255,253) )
+//     For a given N (where N = 255 - #Roots), the number of consecutive recoverable blocks is:
+//       ceiling(# of 4k blocks in disk / (N)) * (255-N)
+//       ie for 2GiB disk: ceiling(524288 / 253) * (255-253) = 2073 * 2 = 4146 blocks = ~16MiB
+// - RootHashSignatureEnable: Validate the root hash against a key stored in the kernel's
+//     system keyring. The signature file should be called "<Name>.p7" and must be stored in
+//     the initramfs. This signature WILL NOT BE included automatically in the initramfs. It must
+//     be included via an out of band build step.
```

Comment:
Do we have an example of the out-of-band build step required for injecting the <Name>.p7 file into the initramfs?